### PR TITLE
naptan get stops lambda function

### PR DESCRIPTION
**Pull request by Danielleoptibus**

### CODE REVIEW

File: naptan_create/create_table.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+COLUMNS = ['ATCOCode', 'NaptanCode', 'PlateCode', 'CleardownCode',
+           'CommonName', 'CommonNameLang', 'ShortCommonName', 'ShortCommonNameLang',
+           'Landmark', 'LandmarkLang', 'Street', 'StreetLang', 'Crossing',
+           'CrossingLang', 'Indicator', 'IndicatorLang', 'Bearing', 'NptgLocalityCode',
+           'LocalityName', 'ParentLocalityName', 'GrandParentLocalityName', 'Town',
+           'TownLang', 'Suburb', 'SuburbLang', 'LocalityCentre', 'GridType', 'Easting',
+           'Northing', 'Longitude', 'Latitude', 'StopType', 'BusStopType',
+           'TimingStatus', 'DefaultWaitTime', 'Notes', 'NotesLang',
+           'AdministrativeAreaCode', 'CreationDateTime', 'ModificationDateTime',
+           'RevisionNumber', 'Modification', 'Status']
```
Comment (urishalit) [inline naptan_create/create_table.py]: I think this isn't needed, we can instead put it dynamically from the file

File: naptan_create/create_table.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+def lambda_handler():
```
Comment (urishalit) [inline naptan_create/create_table.py]: As this won't be a  lambda after all, let's change this

File: naptan_create/create_table.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    url = urlopen("http://naptan.app.dft.gov.uk/DataRequest/Naptan.ashx?format=csv")
```
Comment (urishalit) [inline naptan_create/create_table.py]: Let's change the url to be in a constant var

File: naptan_create/create_table.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    # download and unzip file, save Stops.csv in tmp file
+    try:
+        with ZipFile(BytesIO(url.read())) as my_zip_file:
```
Comment (urishalit) [inline naptan_create/create_table.py]: rename `my_zip_file`

File: naptan_create/create_table.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            for contained_file in my_zip_file.namelist():
+                with open("/tmp/stops_temp_file.csv", "wb") as output:
+                    if contained_file == 'Stops.csv':
```
Comment (urishalit) [inline naptan_create/create_table.py]: Let's change this to just use the `stops.csv` file you know where you extracted the file, so you should know the path to it.

File: naptan_create/create_table.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+                        for line in my_zip_file.open(contained_file).readlines():
+                            output.write(line)
+                        break
+    except Exception:
+        print("cannot download Naptan data from url")
+        raise
+
+    # Connect to DynamoDB using boto
+    dynamo_db = boto3.resource('dynamodb', region_name='eu-west-1')
```
Comment (urishalit) [inline naptan_create/create_table.py]: Change the region to be a constant as well.

File: naptan_create/create_table.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+    # Connect to the DynamoDB table
+    table = dynamo_db.Table('Naptan')
+
+    # read temp file and add each item to dynamodb
+    with table.batch_writer() as batch:
+        try:
+            with codecs.open("/tmp/stops_temp_file.csv", 'r', encoding='ascii', errors='ignore') as stops_csv:
```
Comment (urishalit) [inline naptan_create/create_table.py]: Put a comment and why we open with codecs and ignore errors

File: naptan_create/create_table.py [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+                    # by default, convert empty strings to null
+                    stop = create_stop(row)
+                    batch.put_item(Item=stop)
+        except Exception:
+            print("cannot read from temp file")
+            raise
+
+
+def create_stop(row):
+    stop = {column: row[column] if row[column] != '' else 'null' for column in COLUMNS}
```
Comment (urishalit) [inline naptan_create/create_table.py]: The `DictReader` already returns a dict from key to value, no need for this.

**Review by urishalit — APPROVED** [review]

File: naptan_create/create_table.py (hunk 1 above)
Comment (Danielleoptibus) [reply naptan_create/create_table.py]: done

File: naptan_create/create_table.py (hunk 3 above)
Comment (Danielleoptibus) [reply naptan_create/create_table.py]: DONE

File: naptan_create/create_table.py (hunk 4 above)
Comment (Danielleoptibus) [reply naptan_create/create_table.py]: done

File: naptan_create/create_table.py (hunk 6 above)
Comment (Danielleoptibus) [reply naptan_create/create_table.py]: done

File: naptan_create/create_table.py (hunk 7 above)
Comment (Danielleoptibus) [reply naptan_create/create_table.py]: done

File: naptan_create/create_table.py (hunk 8 above)
Comment (Danielleoptibus) [reply naptan_create/create_table.py]: done

File: naptan_create/create_table.py (hunk 2 above)
Comment (Danielleoptibus) [reply naptan_create/create_table.py]: done

File: naptan_create/create_table.py (hunk 5 above)
Comment (Danielleoptibus) [reply naptan_create/create_table.py]: can you explain face to face? not sure I understand

**Pull request closed without merging**